### PR TITLE
Extract Aspire.Hosting.Azure.Tests

### PR DESCRIPTION
**Pull request by eerhardt**
Contributes to #4294

This also allows us to remove the Aspire.Hosting.Tests.SharedShim, now that VolumeNameGenerator is only found in Aspire.Hosting.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5013)

### CODE REVIEW

File: tests/Aspire.Hosting.Azure.Tests/AzureBicepResourceTests.cs [hunk 1]
```diff
@@ -732,7 +732,7 @@ public async Task WithReferenceAppInsightsSetsEnvironmentVariable()
 
         appInsights.Resource.Outputs["appInsightsConnectionString"] = "myinstrumentationkey";
 
-        var serviceA = builder.AddProject<ProjectA>("serviceA")
+        var serviceA = builder.AddProject<ProjectA>("serviceA", o => o.ExcludeLaunchProfile = true)
```
Comment (radical) [inline tests/Aspire.Hosting.Azure.Tests/AzureBicepResourceTests.cs:735]: Just curious - why is this needed now?

File: tests/Aspire.Hosting.Azure.Tests/AzureSchemaTests.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+using Xunit;
+
+namespace Aspire.Hosting.Azure.Tests;
+
+public class AzureSchemaTests
+{
+    [Fact]
+    public void ValidateApplicationSamples()
+    {
+        new SchemaTests().ValidateApplicationSamples("CdkResourceWithChildResource", (IDistributedApplicationBuilder builder) =>
```
Comment (radical) [inline tests/Aspire.Hosting.Azure.Tests/AzureSchemaTests.cs:14]: Why does this say `CdkResourceWithChildResource`? Will CDK be used here because of `PublishAsAzurePostgresFlexibleServer`?

And is `CDK` - AWS CDK being used for deploying to Azure here?

File: tests/Aspire.Hosting.Azure.Tests/AzureBicepResourceTests.cs (hunk 1 above)
Comment (eerhardt) [reply tests/Aspire.Hosting.Azure.Tests/AzureBicepResourceTests.cs:735]: See https://github.com/dotnet/aspire/issues/4877

File: tests/Aspire.Hosting.Azure.Tests/AzureSchemaTests.cs (hunk 2 above)
Comment (eerhardt) [reply tests/Aspire.Hosting.Azure.Tests/AzureSchemaTests.cs:14]: This is just moving the existing test in:

https://github.com/dotnet/aspire/blob/df935f4ee2e76d088235ea0fe91a0b61405cb3fe/tests/Aspire.Hosting.Tests/Schema/SchemaTests.cs#L110-L114

That test needed to be moved because Aspire.Hosting.Tests is no longer referencing the Aspire.Hosting.Azure libraries. So I moved the test to the Azure tests.

**Review by radical — APPROVED** [review]

**Comment by joperezr** [comment]
@eerhardt no need to backport as release/8.1 branch is not expected to ship assets from it again. `main` branch is currently the 8.1.1 branch. Essentially `release/*` branches are mainly stabilization branches, as the intent is to "ship from `main`" each month. If the content is worthy enough for a minor version bump, then we will, otherwise we'll keep it as a patch update.
